configuration/pih/htmlforms/hiv/section-hiv-history.xml

```diff
@@ -180,6 +180,97 @@
             });
         </script>
     </ifMode>
+    <ifMode mode="VIEW" include="false">
+        <script type="text/javascript">
+            jq(document).ready(function() {
+
+            var x = 1;
+            var elem=document.getElementsByClassName('contacts');
+            var contactSize=elem.length;
+
+            var hideOtherContacts = function() {
+            jq('#contact-2').hide();
+            jq('#contact-3').hide();
+            jq('#contact-4').hide();
+            jq('#contact-5').hide();
+            jq('#contact-6').hide();
+            jq('#contact-7').hide();
+            jq('#contact-8').hide();
+            jq('#contact-9').hide();
+            jq('#contact-10').hide();
+            };
+
+            jq('#contact-'+x).show();
+            hideOtherContacts();
+            jq('#show-less-contacts-button').hide()
+
+
+            jq('#show-more-contacts-button').click(function() {
+            if(contactSize > x){
+            x++;
+            jq('#contact-'+x).show();
+            if(x>1){
+            jq('#show-less-contacts-button').show()
+            }
+            if(x==10){
+            jq('#show-more-contacts-button').hide();
+            }
+
+            }
+            });
+
+            jq('#show-less-contacts-button').click(function() {
+
+            if(contactSize > x || x==10){
+            if(x>1){
+            jq('#contact-'+x).hide();
+            x--;
+            if(x==1){
+            jq('#show-less-contacts-button').hide()
+            }
+            }
+            jq('#show-more-contacts-button').show();
+            }
+
+            });
+
+
+
+            <!-- Phone Number Regex functionality -->
+            for(let i=0; contactSize>i; i++){
+
+            jq("#contact-hiv-phone-"+i).children(":input").change(function(e) {
+            var val=e.target.value;
+            phoneNumber(val,i);
+
+            });
+            }
+
+            function phoneNumber(inputted,index){
+
+            var pattern1= /^\d{8}$/;
+            var pattern2= /^\d{4}(?:\)|[-|\s])?\s*?\d{4}$/;
+
+            if(inputted.match(pattern1) || inputted.match(pattern2)){
+
+            jq("#next").prop("disabled", false)
+            jq("#submit").prop("disabled", false)
+            jq("#contact-phone-error-message-"+index).text('');
+
+            }else{
+
+            jq("#contact-phone-error-message-"+index).text('<uimessage code="pihcore.invalid.phoneNumber"/>');
```

Comment:
This code is added into openmrs-config-pihemr/configuration/messageproperties/messages_en.properties

The French and Haitian translations use Transifex,  @louidorjp I can show you tomorrow?

Reply:
ah ok. no problem